streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -193,6 +196,26 @@ private void closeAndRevive(final Map<Task, Collection<TopicPartition>> taskWith
                 log.error("Error suspending corrupted task {} ", task.id(), swallow);
             }
             task.closeDirty();
+            if (task.isActive()) {
+                // Pause so we won't poll any more records for this task until it has been re-initialized
+                // Note, closeDirty already clears the partitiongroup for the task.
+                final Set<TopicPartition> currentAssignment = mainConsumer().assignment();
+                final Set<TopicPartition> assignedToPauseAndReset =
+                    Utils.intersection(HashSet::new, currentAssignment, task.inputPartitions());
```

Comment:
I don't fully understand how this could happen, but I saw it in several tests, that an active task isn't assigned the input TopicPartition for itself. Whether or not it can happen with a real consumer, or only in tests with the MockConsumer, doesn't really seem important. If the topic isn't assigned, we certainly don't need to reset it.

Reply:
FYI, this is also fixed (better) in https://github.com/apache/kafka/pull/8996

Reply:
I agree that it should never happen. Maybe it's a test setup issue? Should we be worries to mask a potential bug by writing the code like this? Or should we rather fail hard (if we assume it would indicate a bug) and make sure the update the tests?

Reply:
Could this happen due to some race condition with a deleted regex topic? I assume that's not happening in your tests, just wondering if this check actually is necessary or if it would only be masking a bug as Matthias said.

I'm _pretty_ sure we should always go through the process of updating the task's input partitions immediately after updating the consumer's assignment, so we would never find them to be out of sync.  But can the consumer assignment be updated outside of a rebalance, eg based on metadata refresh?

Reply:
Thanks all, it may well have just been a problem with the test setup, but @ableegoldman is making me nervous that there may be some edge cases where it's expected in production code.

Since we would never have had a dependency on the task inputs actually being assigned here, and now we do, I'm concerned that if we try to be strict right now, we may be introducing an IllegalStateException. Normally, this wouldn't be a problem, but this PR is a last-minute hotfix before a release, so I'm feeling cautious.

Since we seem to agree that the code is correct, although it may not be strict, I've added a warning log instead of increasing strictness.